src/dblib_stubs.c

```diff
@@ -243,12 +289,20 @@ value ocaml_freetds_dbopen(value vuser, value vpasswd, value vchar_set,
     }
   }
 
-  dbproc = dbopen(login, String_val(vserver));
+  server = String_val(vserver);
+  caml_release_runtime_system();
+  dbproc = dbopen(login, server);
+  caml_acquire_runtime_system();
```

Comment:
@Chris00 This is the case where we need to handle `have_runtime_lock` being null, since we don't want to hold the lock while we're making the initial connection to the server, but we don't have a (non-null) dbproc until line 302.